src/gpu/intel/jit/ir/tensor.hpp

```diff
@@ -40,112 +41,6 @@ namespace gpu {
 namespace intel {
 namespace jit {
 
-class tensor_t {
```

Comment:
FYI, the DSL really needs to be in its own namespace as it is too easy to hit name conflicts with operations, for example `load` is a very generic name. As such, this rename is not technically necessary as prework, although it still seems like a good idea.

Reply:
What do you think about combining IR/DSL under a single namespace (say `ir`)? It'd be weird to have say `expr_t` in one namespace and put `for_`, `if_`, etc into a different namespace, while they will be always used together in kernel code.

> for example load is a very generic name

To avoid naming clashes - we can rename abstractions. For your example, `load_t` - > `reg_load_t` looks cleaner and more specific.